src/sqlfluff/core/dialects/dialect_snowflake.py

```diff
@@ -479,3 +503,96 @@ class CreateStatementSegment(BaseSegment):
         Sequence("IF", "NOT", "EXISTS", optional=True),
         Ref("ObjectReferenceSegment"),
     )
+
+
+@snowflake_dialect.segment()
+class ShowStatementSegment(BaseSegment):
+    """A snowflake `SHOW` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/show.html
+    """
+
+    _object_types_pural = OneOf(
+        "PARAMETERS",
+        Sequence("GLOBAL", "ACCOUNTS"),
+        "REGIONS",
+        Sequence("REPLICATION", "ACCOUNTS"),
+        Sequence("REPLICATION", "DATABASES"),
+        "PARAMETERS",
+        "VARIABLES",
+        "TRANSACTIONS",
+        "LOCKS",
+        "PARAMETERS",
+        "FUNCTIONS",
+        Sequence("NETWORK", "POLICIES"),
+        "SHARES",
+        "ROLES",
+        "GRANTS",
+        "USERS",
+        "WAREHOUSES",
+        "DATABASES",
+        Sequence(
+            OneOf("API", "NOTIFICATION", "SECURITY", "STORAGE", optional=True),
+            "INTEGRATIONS",
+        ),
+        "SCHEMAS",
+        "OBJECTS",
+        "TABLES",
+        Sequence("EXTERNAL", "TABLES"),
+        "VIEWS",
+        Sequence("MATERIALIZED", "VIEWS"),
+        Sequence("MASKING", "POLICIES"),
+        "COLUMNS",
+        Sequence("FILE", "FORMATS"),
+        "SEQUENCES",
+        "STAGES",
+        "PIPES",
+        "STREAMS",
+        "TASKS",
+        Sequence("USER", "FUNCTIONS"),
+        Sequence("EXTERNAL", "FUNCTIONS"),
+        "PROCEDURES",
+        Sequence("FUTURE", "GRANTS"),
+    )
+
+    _object_scope_types = OneOf(
+        "ACCOUNT",
+        "SESSION",
+        Sequence(
+            OneOf(
+                "DATABASE",
+                "SCHEMA",
+                "TABLE",
+                "TASK",
+                "USER",
+                "WAREHOUSE",
+                "VIEW",
+            ),
+            Ref("ObjectReferenceSegment", optional=True),
+        ),
+    )
+
+    type = "show_statement"
+
+    match_grammar = Sequence(
+        "SHOW",
+        OneOf("TERSE", optional=True),
+        _object_types_pural,
+        OneOf("HISTORY", optional=True),
+        Sequence("LIKE", Ref("ObjectReferenceSegment"), optional=True),
+        Sequence(
+            OneOf("ON", "TO", "OF", "IN"),
+            OneOf(
+                Sequence(_object_scope_types),
+                Ref("ObjectReferenceSegment"),
+            ),
+            optional=True,
+        ),
```

Comment:
Would you be able to add test cases (examples from snowflake docs is probably best) to cover all the logic here, and link to the snowflake documentation in each case?